spec/controllers/plan_wizard_controller_spec.rb

```diff
@@ -0,0 +1,226 @@
+      it "should work" do
+        wp = create(:work_plan, owner_email: @user.email)
+
+        get :show, params: { work_plan_id: wp.id, id: 'set' }
+
+        expect(response).to have_http_status(:ok)
+        expect(response.redirect_url).to be_nil
+        expect(flash[:alert]).to be_nil
+      end
+    end
+
+    context "when the order belongs to another user" do
+      it "should fail authorisation" do
+        user2 = OpenStruct.new(email: 'dirk@sanger.ac.uk', groups: ['world'])
+        wp = create(:work_plan, owner_email: user2.email)
+
+        get :show, params: { work_plan_id: wp.id, id: 'set' }
+
+        expect(response).to have_http_status(:found)
+        expect(response.redirect_url).to be_present
+        expect(flash[:alert]).to match(/not authori[sz]ed/)
+      end
+    end
+  end
+
+  describe "#update" do
+    context "when there is nothing to update" do
+      before do
+        @wp = create(:work_plan, owner_email: @user.email)
+      end
+      context "when work order is at set step" do
+        it "should show error and stay on step when no set is selected" do
+          put :update, params: { work_plan_id: @wp.id, id: 'set'}
+          expect(flash[:error]).to eq 'Please select an option to proceed'
+
+        end
+      end
+      context "when work order is at project step" do
+        it "should show error and stay on step when no project is selected" do
+          put :update, params: { work_plan_id: @wp.id, id: 'project'}
+          expect(flash[:error]).to eq 'Please select an option to proceed'
+          expect(UpdatePlanService).not_to receive(:new)
+          expect(response.redirect_url).to be_nil
+        end
+        it "should show error and stay on step when a project not authorised for the user is selected" do
+          allow(StudyClient::Node).to receive(:authorize!).and_raise(CanCan::AccessDenied)
+          put :update, params: { work_plan_id: @wp.id, work_plan: { project_id: 1234 }, id: 'project'}
+          expect(UpdatePlanService).not_to receive(:new)
+          expect(response.redirect_url).to be_nil
+        end
+      end
+      context "when work order is at product step" do
+        it "should show error and stay on step when product is empty" do
+          put :update, params: { work_plan_id: @wp.id, id: 'product', work_plan: {comment:"", desired_date:"", product_id:""}}
+          expect(flash[:error]).to eq 'Please select a set in an earlier step.'
+          expect(UpdatePlanService).not_to receive(:new)
+          expect(response.redirect_url).to be_nil
+        end
+
+        it "should show error and stay on step when no product is selected" do
+          put :update, params: { work_plan_id: @wp.id, id: 'product', work_plan: {comment:"", desired_date:""}}
+          expect(flash[:error]).to eq 'Please select a set in an earlier step.'
+          expect(UpdatePlanService).not_to receive(:new)
+          expect(response.redirect_url).to be_nil
+        end
+
```

Comment:
I dont think the error here should be referencing sets. Add a original_set_uuid for the work plan, if you are testing there is no product selected

Reply:
Yes, it was not right. I've changed it